File: totosuki/ABC/C/129.py
N, M = map(int, input().split())
A = {int(input()) for _ in range(M)}
MOD = 1000000007

dp = [1] + [0] * N

for i in range(1, N+1):
  if i in A: continue
  dp[i] = (dp[i-1] + dp[i-2]) % MOD

print(dp[N])

File: OCH_Tao/ABC/B/129.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const N = parseInt(input[0], 10);
const W = (input[1].split(" ")).map(x => parseInt(x, 10));
let cnt = [];
for (let i = 1; i < N; i++) {
  const X = [...W.slice(0, i)].reduce((a, b) => a + b, 0);
  const Y = [...W.slice(i)].reduce((a, b) => a + b, 0);
  cnt.push(Math.abs(X - Y));
}
console.log(Math.min(...cnt));